### PR TITLE
Upgrade to Go 1.18

### PR DESCRIPTION
**Pull request by epk**
Signed-off-by: Aditya Sharma <git@adi.run>

### Description of your changes

- Upgrades Go to 1.18
- Fixes some minor lint errors

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: pkg/fieldpath/paved.go [hunk 1]
```diff
@@ -38,15 +38,15 @@ func (e errNotFound) IsNotFound() bool {
 // index was out of bounds in an array.
 func IsNotFound(err error) bool {
 	cause := errors.Cause(err)
-	_, ok := cause.(interface {
+	_, ok := cause.(interface { //nolint: errorlint // Skip errorlint for interface type
```
Comment (muvaf) [inline pkg/fieldpath/paved.go:41]: Is there no way to use `any` here as well?

File: pkg/parser/parser.go [hunk 2]
```diff
@@ -102,10 +102,10 @@ func (p *PackageParser) Parse(ctx context.Context, reader io.ReadCloser) (*Packa
 	do := json.NewSerializerWithOptions(json.DefaultMetaFactory, p.objScheme, p.objScheme, json.SerializerOptions{Yaml: true})
 	for {
 		bytes, err := yr.Read()
-		if err != nil && err != io.EOF {
+		if err != nil && !errors.Is(err, io.EOF) {
 			return pkg, err
 		}
-		if err == io.EOF {
+		if errors.Is(err, io.EOF) {
```
Comment (muvaf) [inline pkg/parser/parser.go:108]: Just checked that `errors.Is` returns the same answer for `== io.EOF` by running the tests with coverage and seeing it go into this statement.

**Review by muvaf — APPROVED** [review]
Thanks @epk !

File: pkg/fieldpath/paved.go (hunk 1 above)
Comment (epk) [reply pkg/fieldpath/paved.go:41]: `any` is only explicitly interoperable with `interface{}`, here it is interface with function definitions

Comment (muvaf) [reply pkg/fieldpath/paved.go:41]: Do you know if there is a plan to make it available for use in these cases as well, or are we stuck with `interface{}` for the foreseeable future?

**Pull request merged**